File: src/main/java/leetCode/offer/Offer58_1.java
package leetCode.offer;

/**
 * @author jingxinwu
 * @date 2021-11-25 1:13 上午
 */
public class Offer58_1 {

    public String reverseWords(String s) {
        String[] strs = s.trim().split(" "); // 删除首尾空格，分割字符串
        StringBuilder res = new StringBuilder();
        for (int i = strs.length - 1; i >= 0; i--) { // 倒序遍历单词列表
            if (strs[i].equals("")) {
                continue; // 遇到空单词则跳过
            }
            res.append(strs[i]).append(" "); // 将单词拼接至 StringBuilder
        }
        return res.toString().trim(); // 转化为字符串，删除尾部空格，并返回
    }
}

File: go/src/leetcode/offer/Offer58_1.go
package offer

import "strings"

/**
 * @author wujingxinit@outlook.com
 * @date 2023/1/3 20:43
 */

func reverseWords(s string) string {
	fields := strings.Fields(s)
	for i, j := 0, len(fields)-1; i < j; i, j = i+1, j-1 {
		fields[i], fields[j] = fields[j], fields[i]
	}
	return strings.Join(fields, " ")
}
